lib/internal/http2/core.js

```diff
@@ -200,33 +231,89 @@ function onSessionShutdownComplete(status, wrap) {
+// because the onSelectPadding is called once for each frame, which
+// means for each frame there's a call across the C++/JS boundary and
+// a trip through EventEmitter land. This likely is not going to be
+// the most efficient padding strategy to use.
 function onSelectPadding(frameLen, maxFramePayloadLen) {
   const owner = this[kOwner];
   const ret = {length: frameLen};
   owner.emit('selectPadding', frameLen, maxFramePayloadLen, ret);
   return Math.min(maxFramePayloadLen, Math.max(frameLen, ret.length | 0));
 }
 
+// Called when the socket is connected to handle a pending request.
 function requestOnConnect(headers, options) {
   const session = this[kSession];
   const state = session[kState];
   const streams = state.streams;
   const handle = session[kHandle];
+  // ret will be either the reserved stream ID (if positive)
+  // or an error code (if negative)
   const ret = handle.submitRequest(mapToHeaders(headers),
-                                   Boolean(options.endStream),
+                                   !!options.endStream,
                                    options.parent | 0,
                                    options.weight | 0,
-                                   Boolean(options.exclusive));
+                                   !!options.exclusive);
   if (emitErrorIfNecessary(this, ret))
     return;
   this[kInit](ret);
   streams.set(ret, this);
 }
 
+function validatePriorityOptions(options) {
+  if (options.weight === undefined)
+    options.weight = NGHTTP2_DEFAULT_WEIGHT;
+  else if (typeof options.weight !== 'number') {
+    const err = new errors.RangeError('ERR_INVALID_OPT_VALUE',
+                                      'weight',
+                                      options.weight);
+    Error.captureStackTrace(err, validatePriorityOptions);
+    throw err;
+  }
+  if (options.parent === undefined)
+    options.parent = 0;
+  else if (typeof options.parent !== 'number' || options.parent < 0) {
+    const err = new errors.RangeError('ERR_INVALID_OPT_VALUE',
+                                      'parent',
+                                      options.parent);
+    Error.captureStackTrace(err, validatePriorityOptions);
+    throw err;
+  }
+  if (options.exclusive === undefined)
+    options.exclusive = false;
+  else if (typeof options.exclusive !== 'boolean') {
+    const err = new errors.RangeError('ERR_INVALID_OPT_VALUE',
+                                      'exclusive',
+                                      options.exclusive);
+    Error.captureStackTrace(err, validatePriorityOptions);
+    throw err;
+  }
+  if (options.silent === undefined)
+    options.silent = false;
+  else if (typeof options.silent !== 'boolean') {
+    const err = new errors.RangeError('ERR_INVALID_OPT_VALUE',
+                                      'silent',
+                                      options.silent);
+    Error.captureStackTrace(err, validatePriorityOptions);
+    throw err;
+  }
+}
```

Comment:
this is not really readable, I would prefer a blank line between the if blocks, and braces everywhere.